db/compaction/compaction_job.h

```diff
@@ -285,6 +299,15 @@ struct CompactionServiceResult {
   uint64_t bytes_read;
   uint64_t bytes_written;
   CompactionJobStats stats;
+
+  // serialization interface to read and write the object
+  static Status Read(const std::string& data_str, CompactionServiceResult* obj);
+  Status Write(std::string* output);
```

Comment:
Should Write take a BinaryFormatVersion of what format to write?

Reply:
We can add that if we have other format versions.